akka-actor/src/main/scala/akka/routing/MetricsBasedResizer.scala

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.routing
+
+import java.time.LocalDateTime
+
+import scala.collection.immutable
+import scala.concurrent.forkjoin.ThreadLocalRandom
+import scala.concurrent.duration._
+
+import com.typesafe.config.Config
+
+import akka.actor._
+import akka.util.JavaDurationConverters._
+
+import MetricsBasedResizer._
+
+trait MetricsBasedResizer extends Resizer {
```

Comment:
I think we should come up with a better name that is somewhat more specific. @ktoso do you have any name suggestion for this thing?